linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -5677,28 +5691,35 @@ private[optimizer] object OptimizerCore {
       RecordType(recordFields)
     }
 
-    private val recordFieldNames: Map[FieldName, RecordType.Field] = {
+    private val recordFields: Map[FieldName, RecordType.Field] = {
       val elems = for ((fieldDef, recordField) <- allFields.zip(recordType.fields))
         yield fieldDef.name.name -> recordField
       elems.toMap
     }
 
+    private[OptimizerCore] def findField(fieldName: FieldName): RecordType.Field =
+      recordFields(fieldName)
+
     private[OptimizerCore] def fieldOriginalName(fieldName: FieldName): OriginalName =
-      recordFieldNames(fieldName).originalName
+      recordFields(fieldName).originalName
 
     override def equals(that: Any): Boolean = that match {
```

Comment:
When do we need this? For incremental processing?

If yes, consider converting to a special purpose equality method (just like `sameClassAs` is). Because IIUC in the context of `OptimizerCore`, `x.sameClassAs(y)` implies `x == y`.

OTOH, for incremental procesing, checking that `className` equals is bogus.

Reply:
It is for incremental processing. And indeed, for that use case, we must always have the same `className`, and we don't need `hashCode`.

I renamed this one to `sameStructureAs`.